src/cocotb/regression.py

```diff
@@ -430,10 +435,21 @@ def _execute(self, *, sim_failed: bool = False) -> None:
             else:
                 self._test_start_sim_time = get_sim_time("ns")
                 self._test_start_time = time.time()
-                return cocotb._scheduler._add_test(self._test_task)
+
+                if self._first_test:
+                    self._first_test = False
+                    return self._schedule_next_test()
+                else:
+                    return self._timer1._prime(self._schedule_next_test)
 
         return self._tear_down()
 
+    def _schedule_next_test(self, trigger: Optional[Trigger] = None) -> None:
+        if trigger is not None:
+            cocotb._scheduler_inst._sim_phase = cocotb._scheduler.SimPhase.NORMAL
```

Comment:
I feel like I want to see `SimPhase` moved to a global and tracked by the `GPITrigger`s so we don't have to do this here.